pkgs/applications/misc/qmapshack/default.nix

```diff
@@ -1,12 +1,15 @@
-{ mkDerivation, lib, fetchurl, fetchpatch, cmake, qtscript, qtwebengine, gdal, proj, routino, quazip }:
+{ mkDerivation, lib, fetchFromGitHub, cmake
+, qtscript, qtwebengine, gdal, proj, routino, quazip }:
 
 mkDerivation rec {
   pname = "qmapshack";
-  version = "1.13.1";
+  version = "1.13.2";
 
-  src = fetchurl {
-    url = "https://bitbucket.org/maproom/qmapshack/downloads/${pname}-${version}.tar.gz";
-    sha256 = "15x1b2q0hr1vx006f9hjc4cvfjvxvfdwybw32qvczdyc3crq0mc9";
+  src = fetchFromGitHub {
+    owner = "Maproom";
+    repo = pname;
+    rev = "763cfc149566325cce9e4690cb7b5f986048f86a";
```

Comment:
We can't use `V_${version}` here with this release, see https://github.com/Maproom/qmapshack/issues/3.

Reply:
From the description of the issue I do not understand why using the tag `V_${version}` is not supposed to work.

Reply:
Some folders required for build are missed from released tarball, those missed folders was added within additional commits just after tagged (V_${version}) commit.

Reply:
How many commits are involved in that? Are they directly on top of the version tag? If it is not to many I would prefer to use `fetchpatch` and `V_${version}`.

Reply:
Well, there are 3 commits:
```nix
    (fetchpatch { # [QMS-3] Add QMapTool from former sub-repo
      url = "https://github.com/Maproom/qmapshack/commit/c6e5240ec25b91d3320f265346a904b5bd906363.patch";
      sha256 = "1fi3jzik4h3q9irrw4lg65am15rfskv2969avazp54wjcrsfbk1r";
    })
    (fetchpatch { # [QMS-3] Add qmt_map2jnx from former sub-repo
      url = "https://github.com/Maproom/qmapshack/commit/79a266943a40bee8fa5e71776c6a76c4d46bfbf8.patch";
      sha256 = "0mgdcjnzb3bw11ybhzfr5w3afrfcl1wvrj48yzapd220dihpsf7d";
    })
    (fetchpatch { # [QMS-3] Add qmt_rgb2pct from former sub-repo
      url = "https://github.com/Maproom/qmapshack/commit/763cfc149566325cce9e4690cb7b5f986048f86a.patch";
      sha256 = "14xiazsysywnp76253alkgmsk1xcxm6bjhxsgf0if6fvg0l1f7cb";
    })
```
But the problem is that the first commit contains .png images, but those images are missed from the .patch above. It looks like GitHub strips binary files from patch.

Reply:
The missing png is a mystery to me. It is not even shown in the commit on GitHub.

If I understand it correctly that means that their release tarball is essentially broken? Looking at the commits there was a lot of things going on related to the move from bitbucket to GitHub. Wouldn't it be easier to ask the developers to make a another release (with a fully functional tarball)?

Reply:
> If I understand it correctly that means that their release tarball is essentially broken?

Yes, it was OK on Bitbucket, but broken on GitHub.

> Wouldn't it be easier to ask the developers to make a another release (with a fully functional tarball)?

I've asked qmapshack maintainer to make a new release, but he's unlikely to want to do it.

Another option here is to use "2019-09-12" version name instead of "1.13.2". So we can use proper 763cfc149566325cce9e4690cb7b5f986048f86a commit without patches.

Reply:
The response is:

> Don't think this is worth the effort of getting into the pressure to release something just because of an outdated source archive.

Reply:
:/ but thanks for asking. To keep it simple, how about this:
Let's just stick with calling it "1.13.2" and use commit 763cfc but let's also add a comment, which clearly states what is going on here. This will be hopefully resolved in the next release.

Reply:
Comment added.